test/unit/test_sign.py

```diff
@@ -148,3 +149,27 @@ def test_sign_prehashed(staging):
     verifier.verify(input_, materials=materials, policy=UnsafeNoOp())
     # verifying against the prehash also works
     verifier.verify(hashed, materials=materials, policy=UnsafeNoOp())
+
+
+@pytest.mark.online
+@pytest.mark.ambient_oidc
+def test_sign_dsse(staging):
+    sign_ctx, _, identity = staging
+
+    ctx = sign_ctx()
+    stmt = (
+        _StatementBuilder()
+        .subjects(
+            [_Subject(name="null", digest={"sha256": hashlib.sha256(b"").hexdigest()})]
+        )
+        .predicate_type("https://cosign.sigstore.dev/attestation/v1")
+        .predicate(
+            {
+                "Data": "",
+                "Timestamp": "2023-12-07T00:37:58Z",
+            }
+        )
+    ).build()
```

Comment:
what's the deal with this `(...).build()`  structure? Looks weird

Reply:
Yeah, that's how Python wants it -- I think there are a few other valid formats, but `ruff` and `black` both favor that one